### PR TITLE
Polls Functionality

### PR DESCRIPTION
**Pull request by Fireworks**
Sorry this PR is messy, and might be hard to follow.  Really just pay attention to changes in `Poll.java` and `Connection.java`.  Make sure you re-disable any bot listeners you do not want active.

**Upgrade to Pircxbot 2.1 if you are going to keep the new server connection strategy with `addServer` and `setServerPassword`.  Otherwise keep the current strategy of `setServer` as it is now.**

- Adds basic "Poll" functionality
    - `!poll option1 option2 option3` (space separated options)
    - `!poll` to end the poll
- Remove a bunch of junk files that shouldn't be tracked in git

### CODE REVIEW

**Comment by amccarthy1** [comment]
I tested this out and looked at the code and it turns out anyone can vote any number of times, might want to add a HashSet of user IDs or something to make sure people can only vote once.

**Comment by Fireworks** [comment]
@amccarthy1 Hah, thanks.   I wrote this late at night... what a huge oversight.  I'll do something like that.

File: src/Poll.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+			int voteKey = 0;
+			try {
+			  voteKey = Integer.parseInt(event.getMessage());
+			} catch (NumberFormatException e) {
+			  // not a valid vote
+			}
+
+			if (voteKey != 0) {
+				votes[voteKey-1] = votes[voteKey-1] + 1;
```
Comment (amccarthy1) [inline src/Poll.java]: This throws an ArrayIndexOutOfBoundsException for negative values or values greater than the number of options (which happens quite a lot in twitch chat). The exception is caught and not logged or anything but you should still probably do bounds checking to handle that a little more gracefully.

**Comment by Fireworks** [comment]
@amccarthy1 Fixed in latest commit.  Went with another HashMap for the user votes, similar to what I am doing in `Predictions` (except storing user UUIDs instead of Nick as key).

Also changed the input matching to split on commas, which should make it easier to do something like: `!poll super mario 64, sunshine, sims`

Let me know if you see any other issues, and appreciate the review.  Should be a lot better now, though I'm still not testing thoroughly :P
